From 4741b2aa78472ec3554cc4b16b151aea59c1f6db Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:36:07 -0400
Subject: [ethernet] i40e: remove dead code

Message-id: <1409658042-22689-7-git-send-email-sassmann@redhat.com>
Patchwork-id: 90200
O-Subject: [RHEL7.1 PATCH 006/281] i40e: remove dead code
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Mitch Williams <mitch.a.williams@intel.com>

This looks like a cut and paste error. The code makes no sense where
it is, and accomplishes nothing. Since we've removed the goto, we can
also get rid of the extraneous brackets.

Change-ID: I9315e3eafeee0a5713c94b0dc57b58b60a849124
Signed-off-by: Mitch Williams <mitch.a.williams@intel.com>
Signed-off-by: Jesse Brandeburg <jesse.brandeburg@intel.com>
Tested-by: Sibai Li <sibai.li@intel.com>
Signed-off-by: Aaron Brown <aaron.f.brown@intel.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit fd1646ee3a311362d3315250b30ccc688b919aef)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_virtchnl_pf.c b/drivers/net/ethernet/intel/i40e/i40e_virtchnl_pf.c
index b9d1c1c..299372b 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_virtchnl_pf.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_virtchnl_pf.c
@@ -408,18 +408,10 @@ static int i40e_alloc_vsi_res(struct i40e_vf *vf, enum i40e_vsi_type type)
      "Could not allocate VF broadcast filter\n");
  }
 
- if (!f) {
-  dev_err(&pf->pdev->dev, "Unable to add ucast filter\n");
-  ret = -ENOMEM;
-  goto error_alloc_vsi_res;
- }
-
  /* program mac filter */
  ret = i40e_sync_vsi_filters(vsi);
- if (ret) {
+ if (ret)
   dev_err(&pf->pdev->dev, "Unable to program ucast filters\n");
-  goto error_alloc_vsi_res;
- }
 
 error_alloc_vsi_res:
  return ret;
-- 
1.7.1